routes/users.go

```diff
@@ -203,3 +218,83 @@ func GetUserHandler(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	json.NewEncoder(w).Encode(user)
 }
+
+func PutUpdateUserHandler(w http.ResponseWriter, r *http.Request) {
+
+	// Checks for food truck ID
+	params := mux.Vars(r)
+	userID, userIDExists := params["userID"]
+	if !userIDExists {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Get user from context
+	_, userLoggedIn := r.Context().Value(middleware.UserKey).(string)
+
+	// Check for a user
+	if !userLoggedIn {
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+
+	userDecoder := json.NewDecoder(r.Body)
+	userDecoder.DisallowUnknownFields()
+
+	// Decode request
+	var updatedUser updateUserRequest
+	err := userDecoder.Decode(&updatedUser)
+	if err != nil {
+		log.Printf("ERROR: %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Determine which fields should be updated
+	var updateData bson.D
+
+	if updatedUser.NameFirst != nil {
+		updateData = append(updateData, bson.E{"firstName", *updatedUser.NameFirst})
+	}
+	if updatedUser.NameLast != nil {
+		updateData = append(updateData, bson.E{"lastName", *updatedUser.NameLast})
+	}
+	if updatedUser.Email != nil {
+		updateData = append(updateData, bson.E{"email", *updatedUser.Email})
+	}
+	if updatedUser.PhoneNumber != nil {
+		updateData = append(updateData, bson.E{"phoneNumber", *updatedUser.PhoneNumber})
+	}
+	if updatedUser.City != nil {
+		updateData = append(updateData, bson.E{"city", *updatedUser.City})
+	}
+	if updatedUser.State != nil {
+		updateData = append(updateData, bson.E{"state", *updatedUser.State})
+	}
+	if updatedUser.DateOfBirth != nil {
+		updateData = append(updateData, bson.E{"dateOfBirth", *updatedUser.DateOfBirth})
+	}
+	if updatedUser.OwnedFoodTrucks != nil {
+		updateData = append(updateData, bson.E{"ownedFoodTrucks", updatedUser.OwnedFoodTrucks})
+	}
+	if updatedUser.Favorites != nil {
+		updateData = append(updateData, bson.E{"favorites", updatedUser.Favorites})
+	}
+	if updatedUser.Reviews != nil {
+		updateData = append(updateData, bson.E{"reviews", updatedUser.Reviews})
+	}
+
+	// Update food truck document
+	update := bson.D{
+		{"$set", updateData},
+	}
+
+	_, err = Db.Collection("users").UpdateOne(r.Context(), queries.WithID(userID), update)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
replace with log.Printf and send a http.StatusInternalServerError so if a db error happens (which is def possible) it doesn't crash the server